File: backend/app/routes/finance.py
from __future__ import annotations

from datetime import datetime
from fastapi import APIRouter, Depends, HTTPException, Response
from sqlalchemy import func
from sqlalchemy.orm import Session

from .. import models, schemas
from ..database import get_db
from ..services.pricing import apply_pricing_to_reservation, mark_paid
from ..services.admin_auth import require_admin_session
from .reservations import _reservation_to_out

router = APIRouter(prefix="/finance", tags=["finance"], dependencies=[Depends(require_admin_session)])

PAID_PAYMENT_STATUSES = {"paid"}
PAYOUT_STATUSES = {"not_ready", "pending", "paid", "blocked", "commission_due", "invoice_sent", "commission_paid"}


def _money(value) -> float:
    return round(float(value or 0), 2)


def _base_paid_query(db: Session):
    return db.query(models.Reservation).join(models.Product).filter(models.Reservation.payment_status == "paid")


@router.get("/summary", response_model=dict)
def finance_summary(db: Session = Depends(get_db)):
    total_reservations = db.query(models.Reservation).count()
    paid_q = db.query(models.Reservation).filter(models.Reservation.payment_status == "paid")
    pending_q = db.query(models.Reservation).filter(models.Reservation.payment_status == "payment_pending")
    refunded_q = db.query(models.Reservation).filter(models.Reservation.payment_status == "refunded")
    pickup_q = db.query(models.Reservation).filter(models.Reservation.payment_status == "pay_on_pickup")
    commission_due_q = pickup_q.filter(models.Reservation.seller_payout_status == "commission_due")

    payout_pending = paid_q.filter(models.Reservation.seller_payout_status == "pending")
    payout_paid = paid_q.filter(models.Reservation.seller_payout_status == "paid")
    payout_blocked = db.query(models.Reservation).filter(models.Reservation.seller_payout_status == "blocked")

    by_payment_status = dict(
        db.query(models.Reservation.payment_status, func.count(models.Reservation.id))
        .group_by(models.Reservation.payment_status).all()
    )
    by_payout_status = dict(
        db.query(models.Reservation.seller_payout_status, func.count(models.Reservation.id))
        .group_by(models.Reservation.seller_payout_status).all()
    )

    return {
        "reservations_total": total_reservations,
        "paid_count": paid_q.count(),
        "payment_pending_count": pending_q.count(),
        "refunded_count": refunded_q.count(),
        "paid_turnover": _money(paid_q.with_entities(func.coalesce(func.sum(models.Reservation.payable_amount), 0)).scalar()),
        "platform_fee_total": _money(paid_q.with_entities(func.coalesce(func.sum(models.Reservation.platform_fee_amount), 0)).scalar()),
        "seller_net_total": _money(paid_q.with_entities(func.coalesce(func.sum(models.Reservation.seller_net_amount), 0)).scalar()),
        "pay_on_pickup_count": pickup_q.count(),
        "pay_on_pickup_turnover": _money(pickup_q.with_entities(func.coalesce(func.sum(models.Reservation.payable_amount), 0)).scalar()),
        "commission_due_count": commission_due_q.count(),
        "commission_due_total": _money(commission_due_q.with_entities(func.coalesce(func.sum(models.Reservation.platform_fee_amount), 0)).scalar()),
        "pending_payout_count": payout_pending.count(),
        "pending_payout_amount": _money(payout_pending.with_entities(func.coalesce(func.sum(models.Reservation.seller_net_amount), 0)).scalar()),
        "paid_payout_count": payout_paid.count(),
        "paid_payout_amount": _money(payout_paid.with_entities(func.coalesce(func.sum(models.Reservation.seller_net_amount), 0)).scalar()),
        "blocked_payout_count": payout_blocked.count(),
        "by_payment_status": by_payment_status,
        "by_payout_status": by_payout_status,
    }


@router.get("/seller-settlements", response_model=list[dict])
def seller_settlements(db: Session = Depends(get_db)):
    stores = db.query(models.Store).order_by(models.Store.name.asc()).all()
    result = []
    for store in stores:
        q = db.query(models.Reservation).join(models.Product).filter(models.Product.store_id == store.id)
        paid = q.filter(models.Reservation.payment_status == "paid")
        pending_payout = paid.filter(models.Reservation.seller_payout_status == "pending")
        paid_payout = paid.filter(models.Reservation.seller_payout_status == "paid")
        blocked_payout = q.filter(models.Reservation.seller_payout_status == "blocked")
        commission_due = q.filter(models.Reservation.seller_payout_status == "commission_due")
        if q.count() == 0 and not store.verified:
            continue
        result.append({
            "store_id": store.id,
            "store_name": store.name,
            "city": store.city,
            "reservations_total": q.count(),
            "paid_count": paid.count(),
            "paid_turnover": _money(paid.with_entities(func.coalesce(func.sum(models.Reservation.payable_amount), 0)).scalar()),
            "platform_fee_total": _money(paid.with_entities(func.coalesce(func.sum(models.Reservation.platform_fee_amount), 0)).scalar()),
            "seller_net_total": _money(paid.with_entities(func.coalesce(func.sum(models.Reservation.seller_net_amount), 0)).scalar()),
            "pending_payout_count": pending_payout.count(),
            "pending_payout_amount": _money(pending_payout.with_entities(func.coalesce(func.sum(models.Reservation.seller_net_amount), 0)).scalar()),
            "paid_payout_count": paid_payout.count(),
            "paid_payout_amount": _money(paid_payout.with_entities(func.coalesce(func.sum(models.Reservation.seller_net_amount), 0)).scalar()),
            "blocked_payout_count": blocked_payout.count(),
            "commission_due_count": commission_due.count(),
            "commission_due_total": _money(commission_due.with_entities(func.coalesce(func.sum(models.Reservation.platform_fee_amount), 0)).scalar()),
        })
    return result


def _closeout_rows(db: Session) -> list[dict]:
    stores = db.query(models.Store).order_by(models.Store.name.asc()).all()
    rows = []
    for store in stores:
        q = db.query(models.Reservation).join(models.Product).filter(models.Product.store_id == store.id)
        if q.count() == 0 and not store.verified:
            continue
        paid = q.filter(models.Reservation.payment_status == "paid")
        pickup = q.filter(models.Reservation.payment_status == "pay_on_pickup")
        commission_due = q.filter(models.Reservation.seller_payout_status == "commission_due")
        invoice_sent = q.filter(models.Reservation.seller_payout_status == "invoice_sent")
        commission_paid = q.filter(models.Reservation.seller_payout_status == "commission_paid")
        pending_payout = paid.filter(models.Reservation.seller_payout_status == "pending")
        rows.append({
            "store_id": store.id,
            "store_name": store.name,
            "city": store.city,
            "reservations_total": q.count(),
            "picked_up": q.filter(models.Reservation.status == "picked_up").count(),
            "paid_turnover": _money(paid.with_entities(func.coalesce(func.sum(models.Reservation.payable_amount), 0)).scalar()),
            "pay_on_pickup_turnover": _money(pickup.with_entities(func.coalesce(func.sum(models.Reservation.payable_amount), 0)).scalar()),
            "platform_fee_paid": _money(paid.with_entities(func.coalesce(func.sum(models.Reservation.platform_fee_amount), 0)).scalar()),
            "commission_due": _money(commission_due.with_entities(func.coalesce(func.sum(models.Reservation.platform_fee_amount), 0)).scalar()),
            "commission_due_count": commission_due.count(),
            "invoice_sent": _money(invoice_sent.with_entities(func.coalesce(func.sum(models.Reservation.platform_fee_amount), 0)).scalar()),
            "commission_paid": _money(commission_paid.with_entities(func.coalesce(func.sum(models.Reservation.platform_fee_amount), 0)).scalar()),
            "seller_payout_pending": _money(pending_payout.with_entities(func.coalesce(func.sum(models.Reservation.seller_net_amount), 0)).scalar()),
        })
    return rows


@router.get("/live-closeout", response_model=dict)
def finance_live_closeout(db: Session = Depends(get_db)):
    rows = _closeout_rows(db)
    totals = {
        "reservations_total": sum(r["reservations_total"] for r in rows),
        "picked_up": sum(r["picked_up"] for r in rows),
        "paid_turnover": _money(sum(r["paid_turnover"] for r in rows)),
        "pay_on_pickup_turnover": _money(sum(r["pay_on_pickup_turnover"] for r in rows)),
        "platform_fee_paid": _money(sum(r["platform_fee_paid"] for r in rows)),
        "commission_due": _money(sum(r["commission_due"] for r in rows)),
        "invoice_sent": _money(sum(r["invoice_sent"] for r in rows)),
        "commission_paid": _money(sum(r["commission_paid"] for r in rows)),
        "seller_payout_pending": _money(sum(r["seller_payout_pending"] for r in rows)),
    }
    actions = []
    if totals["commission_due"] > 0:
        actions.append("Pošalji dnevni obračun provizije partnerima sa commission_due iznosom.")
    if totals["seller_payout_pending"] > 0:
        actions.append("Proveri plaćene rezervacije i pripremi isplatu partnerima.")
    if not actions:
        actions.append("Finansijski closeout nema otvorenih kritičnih stavki.")
    return {
        "ok": True,
        "generated_at": datetime.utcnow().isoformat() + "Z",
        "totals": totals,
        "partners": rows,
        "actions": actions,
        "csv": "/finance/live-closeout.csv",
    }


@router.get("/live-closeout.csv")
def finance_live_closeout_csv(db: Session = Depends(get_db)):
    rows = _closeout_rows(db)
    header = [
        "store_id", "store_name", "city", "reservations_total", "picked_up",
        "paid_turnover", "pay_on_pickup_turnover", "platform_fee_paid",
        "commission_due", "commission_due_count", "invoice_sent",
        "commission_paid", "seller_payout_pending",
    ]
    lines = [",".join(header)]
    for row in rows:
        values = [str(row.get(key, "")).replace('"', '""') for key in header]
        lines.append(",".join(f'"{value}"' if "," in value else value for value in values))
    return Response("\n".join(lines) + "\n", media_type="text/csv; charset=utf-8")


@router.patch("/stores/{store_id}/commission-sent", response_model=dict)
def mark_store_commission_sent(store_id: int, reference: str | None = None, db: Session = Depends(get_db)):
    store = db.get(models.Store, store_id)
    if not store:
        raise HTTPException(status_code=404, detail="Partner nije pronađen")
    reservations = db.query(models.Reservation).join(models.Product).filter(
        models.Product.store_id == store_id,
        models.Reservation.seller_payout_status == "commission_due",
    ).all()
    total = 0.0
    for reservation in reservations:
        total += float(reservation.platform_fee_amount or 0)
        reservation.seller_payout_status = "invoice_sent"
        reservation.seller_payout_reference = reference or f"COMMISSION-{store_id}-{datetime.utcnow().strftime('%Y%m%d')}"
        reservation.seller_payout_note = "Dnevni obračun provizije je poslat partneru."
        reservation.updated_at = datetime.utcnow()
    db.commit()
    return {
        "ok": True,
        "store_id": store_id,
        "store_name": store.name,
        "updated_reservations": len(reservations),
        "commission_marked_sent": _money(total),
    }


@router.get("/reservations", response_model=list[schemas.ReservationOut])
def finance_reservations(
    payment_status: str | None = None,
    payout_status: str | None = None,
    store_id: int | None = None,
    limit: int = 200,
    db: Session = Depends(get_db),
):
    query = db.query(models.Reservation).join(models.Product)
    if payment_status:
        query = query.filter(models.Reservation.payment_status == payment_status)
    if payout_status:
        query = query.filter(models.Reservation.seller_payout_status == payout_status)
    if store_id:
        query = query.filter(models.Product.store_id == store_id)
    reservations = query.order_by(models.Reservation.created_at.desc()).limit(limit).all()
    return [_reservation_to_out(r) for r in reservations]


@router.patch("/reservations/{reservation_code}/confirm-ips", response_model=schemas.ReservationOut)
def confirm_ips_payment(reservation_code: str, payload: schemas.FinanceConfirmPaymentRequest, db: Session = Depends(get_db)):
    reservation = db.query(models.Reservation).filter(models.Reservation.reservation_code == reservation_code.upper()).first()
    if not reservation:
        raise HTTPException(status_code=404, detail="Rezervacija nije pronađena")
    if reservation.status in {"cancelled", "expired"}:
        raise HTTPException(status_code=400, detail="Otkazana ili istekla rezervacija ne može da se potvrdi kao plaćena")
    apply_pricing_to_reservation(db, reservation)
    if reservation.payment_status != "paid":
        mark_paid(reservation, provider=payload.provider or "ips_qr", method="IPS QR / ručna potvrda")
    if payload.reference:
        reservation.payment_reference = payload.reference.strip()
    elif not reservation.payment_reference:
        reservation.payment_reference = f"IPS-{reservation.reservation_code}"
    if payload.note:
        reservation.note = ((reservation.note or "") + f"\n[FINANCE] {payload.note.strip()}").strip()
    reservation.seller_payout_status = "pending"
    reservation.updated_at = datetime.utcnow()
    db.commit()
    db.refresh(reservation)
    return _reservation_to_out(reservation)


@router.patch("/reservations/{reservation_code}/payout", response_model=schemas.ReservationOut)
def update_seller_payout(reservation_code: str, payload: schemas.FinancePayoutUpdateRequest, db: Session = Depends(get_db)):
    reservation = db.query(models.Reservation).filter(models.Reservation.reservation_code == reservation_code.upper()).first()
    if not reservation:
        raise HTTPException(status_code=404, detail="Rezervacija nije pronađena")
    if reservation.payment_status != "paid" and payload.seller_payout_status == "paid":
        raise HTTPException(status_code=400, detail="Ne može isplata prodavcu dok rezervacija nije plaćena")
    reservation.seller_payout_status = payload.seller_payout_status
    reservation.seller_payout_reference = payload.reference.strip() if payload.reference else reservation.seller_payout_reference
    reservation.seller_payout_note = payload.note.strip() if payload.note else reservation.seller_payout_note
    if payload.seller_payout_status == "paid":
        reservation.seller_payout_at = datetime.utcnow()
    elif payload.seller_payout_status in {"pending", "blocked", "not_ready", "commission_due", "invoice_sent"}:
        reservation.seller_payout_at = None
    reservation.updated_at = datetime.utcnow()
    db.commit()
    db.refresh(reservation)
    return _reservation_to_out(reservation)

File: backend/app/static/admin/finance.js
const $ = (id) => document.getElementById(id);
const paymentLabels = { unpaid: 'Neplaćeno', payment_pending: 'Plaćanje pokrenuto', paid: 'Plaćeno', refunded: 'Refundirano', failed: 'Neuspešno' };
const payoutLabels = { not_ready: 'Nije spremno', pending: 'Čeka isplatu', paid: 'Isplaćeno', blocked: 'Blokirano', commission_due: 'Provizija za naplatu', invoice_sent: 'Obračun poslat', commission_paid: 'Provizija naplaćena' };

function escapeHtml(str) { return String(str ?? '').replace(/[&<>'"]/g, (c) => ({'&':'&amp;','<':'&lt;','>':'&gt;',"'":'&#39;','"':'&quot;'}[c])); }
function money(amount, currency='RSD') { return `${Number(amount || 0).toLocaleString('sr-RS', { maximumFractionDigits: 2 })} ${currency}`; }
function toast(message) { const el = $('toast'); el.textContent = message; el.classList.add('show'); setTimeout(() => el.classList.remove('show'), 3200); }
function setButtonLoading(btn, text) { const old = btn.innerHTML; btn.disabled = true; btn.innerHTML = `<span class="spinner"></span>${escapeHtml(text)}`; return () => { btn.disabled = false; btn.innerHTML = old; }; }
async function request(url, options={}) { const res = await fetch(url, options); if (!res.ok) { let text = await res.text(); try { text = JSON.parse(text).detail || text; } catch (_) {} throw new Error(text || `HTTP ${res.status}`); } return res.json(); }
function badge(text, cls='') { return `<span class="status ${cls}">${escapeHtml(text)}</span>`; }

function renderStats(data) {
  const items = [
    ['Plaćen promet', money(data.paid_turnover), 'Ukupno kroz aplikaciju'],
    ['Naša provizija', money(data.platform_fee_total), '25% od plaćenih rezervacija'],
    ['Prodavcima neto', money(data.seller_net_total), 'Ukupno neto iznos'],
    ['Čeka isplatu', money(data.pending_payout_amount), `${data.pending_payout_count || 0} rezervacija`],
    ['Plaćene rez.', data.paid_count || 0, 'Status paid'],
    ['IPS u toku', data.payment_pending_count || 0, 'Čeka ručnu potvrdu'],
    ['Isplaćeno', money(data.paid_payout_amount), `${data.paid_payout_count || 0} stavki`],
    ['Blokirano', data.blocked_payout_count || 0, 'Otkazano/refundirano/sporno'],
    ['Plaćanje kod prodavca', money(data.pay_on_pickup_turnover), `${data.pay_on_pickup_count || 0} rezervacija`],
    ['Provizija za naplatu', money(data.commission_due_total), `${data.commission_due_count || 0} stavki`],
  ];
  $('financeStats').innerHTML = items.map(([label, value, sub]) => `<div class="stat-card"><strong>${escapeHtml(value)}</strong><span>${escapeHtml(label)} · ${escapeHtml(sub)}</span></div>`).join('');
}

function renderSellers(rows) {
  $('sellerSettlementsBody').innerHTML = rows.length ? rows.map(s => `
    <tr>
      <td><strong>${escapeHtml(s.store_name)}</strong></td>
      <td>${escapeHtml(s.city || '-')}</td>
      <td>${s.reservations_total}</td>
      <td>${s.paid_count}</td>
      <td>${money(s.paid_turnover)}</td>
      <td>${money(s.platform_fee_total)}</td>
      <td>${money(s.seller_net_total)}</td>
      <td><strong>${money(s.pending_payout_amount)}</strong><br><small>${s.pending_payout_count} stavki</small></td>
      <td>${money(s.paid_payout_amount)}<br><small>${s.paid_payout_count} stavki</small><br><small>Provizija za naplatu: <strong>${money(s.commission_due_total || 0)}</strong></small></td>
    </tr>`).join('') : '<tr><td colspan="9" class="empty">Još nema obračuna za prodavce.</td></tr>';
}

function renderCloseout(data) {
  const t = data.totals || {};
  const items = [
    ['Ukupno rez.', t.reservations_total || 0, 'svi partneri'],
    ['Preuzeto', t.picked_up || 0, 'realizovano'],
    ['Plaćeno online', money(t.paid_turnover), 'kroz aplikaciju'],
    ['Kod partnera', money(t.pay_on_pickup_turnover), 'naplata pri preuzimanju'],
    ['Provizija za naplatu', money(t.commission_due), 'za slanje partnerima'],
    ['Čeka isplatu', money(t.seller_payout_pending), 'neto za partnere'],
  ];
  $('closeoutStats').innerHTML = items.map(([label, value, sub]) => `<div class="stat-card"><strong>${escapeHtml(value)}</strong><span>${escapeHtml(label)} · ${escapeHtml(sub)}</span></div>`).join('');
  const rows = data.partners || [];
  $('closeoutBody').innerHTML = rows.length ? rows.map(row => `
    <tr>
      <td><strong>${escapeHtml(row.store_name)}</strong><br><small>${escapeHtml(row.city || '')}</small></td>
      <td>${row.reservations_total}</td>
      <td>${row.picked_up}</td>
      <td>${money(row.paid_turnover)}</td>
      <td>${money(row.pay_on_pickup_turnover)}</td>
      <td><strong>${money(row.commission_due)}</strong><br><small>${row.commission_due_count || 0} stavki</small></td>
      <td>${money(row.invoice_sent)}</td>
      <td class="actions">
        ${Number(row.commission_due || 0) > 0 ? `<button data-commission-sent="${row.store_id}">Obračun poslat</button>` : '<span class="muted">Nema duga</span>'}
      </td>
    </tr>`).join('') : '<tr><td colspan="8" class="empty">Nema partnera za closeout.</td></tr>';
}

function renderReservations(rows) {
  $('financeReservationsBody').innerHTML = rows.length ? rows.map(r => `
    <tr>
      <td><strong>${escapeHtml(r.reservation_code)}</strong><br><a href="/checkout?code=${encodeURIComponent(r.reservation_code)}">checkout</a></td>
      <td>${escapeHtml(r.product_name || '-') }</td>
      <td>${escapeHtml(r.store_name || '-')}</td>
      <td>${escapeHtml(r.customer_name)}<br><small>${escapeHtml(r.customer_phone)}</small></td>
      <td>${badge(paymentLabels[r.payment_status] || r.payment_status, `pay-${r.payment_status}`)}<br><small>${escapeHtml(r.payment_reference || '-')}</small></td>
      <td>${money(r.payable_amount, r.currency)}</td>
      <td>${money(r.platform_fee_amount, r.currency)}</td>
      <td><strong>${money(r.seller_net_amount, r.currency)}</strong></td>
      <td>${badge(payoutLabels[r.seller_payout_status] || r.seller_payout_status, `payout-${r.seller_payout_status}`)}<br><small>${escapeHtml(r.seller_payout_reference || '')}</small></td>
      <td class="actions">
        ${r.payment_status !== 'paid' ? `<button data-confirm-ips="${escapeHtml(r.reservation_code)}">Potvrdi IPS</button>` : ''}
        ${r.payment_status === 'paid' && r.seller_payout_status !== 'paid' ? `<button data-payout-paid="${escapeHtml(r.reservation_code)}" class="secondary">Isplaćeno</button>` : ''}
        ${r.seller_payout_status !== 'blocked' ? `<button data-payout-block="${escapeHtml(r.reservation_code)}" class="danger">Blokiraj</button>` : ''}
      </td>
    </tr>`).join('') : '<tr><td colspan="10" class="empty">Nema rezervacija za izabrane filtere.</td></tr>';
}

async function loadFinance() {
  const done = setButtonLoading($('refreshFinanceBtn'), 'Osvežavam...');
  try {
    const [summary, sellers] = await Promise.all([request('/finance/summary'), request('/finance/seller-settlements')]);
    renderStats(summary); renderSellers(sellers); await loadCloseout(false); await loadReservations(false);
    $('financeMessage').textContent = 'Finansijski podaci su osveženi.';
  } catch (err) { toast(err.message); } finally { done(); }
}

async function loadCloseout(showToast=true) {
  const data = await request('/finance/live-closeout');
  renderCloseout(data);
  if (showToast) toast('Closeout osvežen');
}

async function markCommissionSent(storeId) {
  const reference = prompt('Referenca obračuna:', `COMMISSION-${storeId}-${new Date().toISOString().slice(0,10).replaceAll('-','')}`);
  if (reference === null) return;
  await request(`/finance/stores/${encodeURIComponent(storeId)}/commission-sent?reference=${encodeURIComponent(reference)}`, { method: 'PATCH' });
}

async function loadReservations(showToast=true) {
  const params = new URLSearchParams();
  const ps = $('paymentStatusFilter').value; const po = $('payoutStatusFilter').value;
  if (ps) params.set('payment_status', ps); if (po) params.set('payout_status', po); params.set('limit', '200');
  const rows = await request(`/finance/reservations?${params}`);
  renderReservations(rows); if (showToast) toast('Rezervacije učitane');
}

async function confirmIps(code, reference='', note='') {
  const body = { provider: 'ips_qr', reference: reference || `IPS-${code}`, note };
  await request(`/finance/reservations/${encodeURIComponent(code)}/confirm-ips`, { method: 'PATCH', headers: { 'Content-Type': 'application/json' }, body: JSON.stringify(body) });
}

async function updatePayout(code, status) {
  const reference = status === 'paid' ? prompt('Referenca isplate prodavcu / broj naloga:', `ISPLATA-${code}`) : '';
  if (status === 'paid' && reference === null) return;
  const note = status === 'blocked' ? prompt('Razlog blokiranja isplate:', 'Spornost / refund / provera') : '';
  if (status === 'blocked' && note === null) return;
  await request(`/finance/reservations/${encodeURIComponent(code)}/payout`, { method: 'PATCH', headers: { 'Content-Type': 'application/json' }, body: JSON.stringify({ seller_payout_status: status, reference, note }) });
}

$('confirmIpsBtn').addEventListener('click', async () => {
  const code = $('financeCodeInput').value.trim().toUpperCase();
  if (!code) return toast('Unesi kod rezervacije');
  const done = setButtonLoading($('confirmIpsBtn'), 'Potvrđujem...');
  try { await confirmIps(code, $('financeReferenceInput').value.trim(), $('financeNoteInput').value.trim()); $('financeMessage').textContent = `Uplata za ${code} je potvrđena. Isplata prodavcu je sada u statusu “čeka isplatu”.`; await loadFinance(); }
  catch (err) { toast(err.message); }
  finally { done(); }
});

$('loadFinanceReservationsBtn').addEventListener('click', () => loadReservations().catch(err => toast(err.message)));
$('loadCloseoutBtn').addEventListener('click', () => loadCloseout().catch(err => toast(err.message)));
$('refreshFinanceBtn').addEventListener('click', loadFinance);
$('closeoutBody').addEventListener('click', async (e) => {
  const storeId = e.target.getAttribute('data-commission-sent');
  if (!storeId) return;
  try {
    await markCommissionSent(storeId);
    await loadFinance();
    toast('Obračun provizije je označen kao poslat');
  } catch (err) {
    toast(err.message);
  }
});
$('financeReservationsBody').addEventListener('click', async (e) => {
  const confirmCode = e.target.getAttribute('data-confirm-ips');
  const paidCode = e.target.getAttribute('data-payout-paid');
  const blockCode = e.target.getAttribute('data-payout-block');
  try {
    if (confirmCode) { if (!confirm(`Potvrditi IPS uplatu za ${confirmCode}?`)) return; await confirmIps(confirmCode); }
    if (paidCode) { await updatePayout(paidCode, 'paid'); }
    if (blockCode) { await updatePayout(blockCode, 'blocked'); }
    await loadFinance(); toast('Status je sačuvan');
  } catch (err) { toast(err.message); }
});

loadFinance();
